README.md

```diff
@@ -1032,6 +1033,57 @@ MongoClient.connect('mongodb://localhost:27017/test', function(err, db) {
 });
 ```
 
+#### <a name="count"></a>count(query, options, callback)
+
+Returns the count of documents that would match a query.
+
+Does not accept options if varyCountByQuery set to true, use countDocuments or estimatedDocumentCount directly instead.
```

Comment:
`countDocuments` and `estimatedDocumentCount` have some common options, so our custom count implementation could safely accept those.

Some options are only accepted by `countDocuments`, we could either ignore them or fall back to `countDocuments` if such option is supplied.

But for the sake of simplicity we could stick to the current implementation and drop options support when custom count implementation is used.

Reply:
In my opinion, the best solution would be to accept `maxTimeMS` option here, while ignoring any other options regardless of which count implementation is used under the hood.

Reply:
With options support it would be hard to tell if it's options or query (both parameters are optional). So I cast away options support for the sake of code simplicity. Besides, options are rarely used.

Reply:
It's not hard at all.

Here is a possible implementation:

```js
Collection.prototype.count = utils.withPromise(function(
	query, options, callback
) {
	if (!this.varyCountByQuery) {
		SourceCollection.prototype.count.apply(this, arguments);
		return;
	}

	if (typeof query === 'function') {
		callback = query;
		query = null;
		options = null;
	} else if (typeof options === 'function') {
		callback = options;
		options = null;
	}
	query = query || null;
	options = options && options.maxTimeMS
		? {maxTimeMS: options.maxTimeMS}
		: {};

	if (query && Object.keys(query).length) {
		SourceCollection.prototype.countDocuments.call(
			this, query, options, callback
		);
	} else {
		SourceCollection.prototype.estimatedDocumentCount.call(
			this, options, callback
		);
	}
});
```

Reply:
It'll undoubtedly work, but it's not clear why we can't use `skip` option, for example. It's kind of misleading behavior. And when `skip` is needed, one should use `countDocuments` anyway. Besides, function signature will look like `count(null, {maxTimeMS: 1000})` for estimated count and it looks a little weird.
So I'd choose more straightforward way and use countDocuments or estimatedDocumentCount when options are needed.

Reply:
You can't call MongoDB's count without first argument when using `options` as well.

Reply:
See for yourself:

```
> await db.rubrics.countDocuments()
637
> await db.rubrics.countDocuments({maxTimeMS: 1000})
0
> await db.rubrics.countDocuments({}, {maxTimeMS: 1000})
637
```

Reply:
I'd rather think about estimatedDocumentCount where condition is not used

Reply:
As for limit and other `countDocuments`-specific options, you can't use them because it's cause undetermined behavior when empty query is supplied.

Reply:
Hm... It seems `estimatedDocumentCount` accepts `limit` and `offset` options despite what official documentation says:

```
> await db.rubrics.estimatedDocumentCount({limit: 100})
100
> await db.rubrics.estimatedDocumentCount({limit: 100, skip: 600})
37
```

Reply:
We could just pass user-supplies options to the underlying driver method then.

Reply:
Ok, added options without any processing